aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/SynchronousNonBlockingStep.java

```diff
@@ -0,0 +1,162 @@
+package org.jenkinsci.plugins.workflow.steps;
+
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+import jenkins.model.Jenkins;
+
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.TestExtension;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import com.google.inject.Inject;
+
+public final class SynchronousNonBlockingStep extends AbstractStepImpl implements Serializable {
+
+    /** State of semaphore steps within one Jenkins home and thus (possibly restarting) test. */
```

Comment:
Probably do not need this complexity, since synchronous steps cannot survive a restart anyway.